### PR TITLE
stuff

### PR DESCRIPTION
**Pull request by armorfid**

### CODE REVIEW

**Comment by winstxnhdw** [comment]
LMFAO, I didn't know our unlimited ammo applied to the enemy as well.

File: lc-hax/Scripts/Patches/UnlimitedAmmoPatch.cs [hunk 1]
```diff
@@ -4,11 +4,11 @@
 
 namespace Hax;
 
-[HarmonyPatch(typeof(ShotgunItem))]
-class UnlimitedAmmoPatch {
-    [HarmonyPatch(nameof(ShotgunItem.ItemActivate))]
-    static void Prefix(ref int ___shellsLoaded) => ___shellsLoaded = 2;
-
-    [HarmonyPatch(nameof(ShotgunItem.ShootGun))]
-    static void Postfix(ref int ___shellsLoaded) => ___shellsLoaded = 2;
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Patches/UnlimitedAmmoPatch.cs:13]: I see you removed the patch on the `ShootGun` method. Would this even work?

**Review by winstxnhdw — CHANGES_REQUESTED** [review]

File: lc-hax/Scripts/Patches/UnlimitedAmmoPatch.cs (hunk 1 above)
Comment (winstxnhdw) [reply lc-hax/Scripts/Patches/UnlimitedAmmoPatch.cs:13]: Hmm.. I see.. `ItemActivate` is used to shoot the gun.

**Comment by winstxnhdw** [comment]
Thanks again, @armorfid!

File: lc-hax/Scripts/Patches/UnlimitedAmmoPatch.cs (hunk 1 above)
Comment (armorfid) [reply lc-hax/Scripts/Patches/UnlimitedAmmoPatch.cs:13]: yes, by setting ammo to 3 before `ItemActivate`, `ShootGun` will decrement it to 2, so all is good

File: lc-hax/Scripts/Modules/StunMod.cs [hunk 2]
```diff
@@ -16,13 +18,20 @@ public sealed class StunMod : MonoBehaviour, IStun {
...
 
         this.Stun(camera.transform.position, 5.0f);
+        int colliders = Physics.OverlapSphereNonAlloc(camera.transform.position, 5.0f, this.Colliders);
+
+        for (int i = 0; i < colliders; i++) {
+            GameObject gameObject = this.Colliders[i].gameObject;
 
-        foreach (RaycastHit raycastHit in Helper.RaycastForward()) {
-            GameObject gameObject = raycastHit.collider.gameObject;
+            if (gameObject.TryGetComponent(out Turret _) && gameObject.TryGetComponent(out Landmine _)) {
```
Comment (armorfid) [inline lc-hax/Scripts/Modules/StunMod.cs:26]: this won't work, should be `!a && !b`

File: lc-hax/Scripts/Modules/StunMod.cs [hunk 3]
```diff
@@ -16,13 +18,20 @@ public sealed class StunMod : MonoBehaviour, IStun {
...
+                return;
+            }
 
-            if (gameObject.TryGetComponent(out Turret _) || gameObject.TryGetComponent(out Landmine _)) {
-                gameObject.GetComponent<TerminalAccessibleObject>()?.CallFunctionFromTerminal();
+            if (!gameObject.TryGetComponent(out TerminalAccessibleObject terminalAccessibleObject)) {
```
Comment (armorfid) [inline lc-hax/Scripts/Modules/StunMod.cs:30]: `?.` is way cleaner than this 😂

File: lc-hax/Scripts/Modules/StunMod.cs (hunk 2 above)
Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/StunMod.cs:26]: Whoops, you're right.

File: lc-hax/Scripts/Modules/StunMod.cs (hunk 3 above)
Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/StunMod.cs:30]: Yeah, but I've been traumatised by Unity's fake null problem.

Comment (armorfid) [reply lc-hax/Scripts/Modules/StunMod.cs:30]: ah ok, I am not familiar with Unity's quirks, first time using it at all

Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/StunMod.cs:30]: Yeah, I didn't know about it until I did this project as well. It was extremely painful to debug.

**Comment by armorfid** [comment]
the edits you made to my jamming code do not work, did you test them in-game before pushing?

**Comment by winstxnhdw** [comment]
I changed the functionality. It only jams in a 5m radius.

**Comment by winstxnhdw** [comment]
Oh, but now that you mentioned, I did make a small mistake that can cause inconsistent behaviour. Let me push a fix for that.

**Comment by armorfid** [comment]
I was within 1m and it did not work

**Comment by winstxnhdw** [comment]
Yeah, I think it was hitting your own collider then early returning each time.

**Comment by armorfid** [comment]
and I do still want to jam them from afar, just like how `M3` works, so I very much prefer raycast either way

**Comment by winstxnhdw** [comment]
I think it should be a new mod then.

**Comment by armorfid** [comment]
I strongly believe stunclick should also have a raycast too in addition to the radius, to stun enemies from afar

**Comment by winstxnhdw** [comment]
Okay, then we should add raycast for the usual stun as well.

**Comment by winstxnhdw** [comment]
Alright, I've pushed that.

**Comment by armorfid** [comment]
Very nice!

**Pull request merged**